boot/boot_serial/src/boot_serial.c

```diff
@@ -267,6 +271,31 @@ bs_list(char *buf, int len)
     boot_serial_output();
 }
 
+static uint32_t next_offset(uint32_t curr_off, uint32_t img_size)
```

Comment:
I would like to have function for checking the stalling timeout explicitly called in bs_uploda().
I think calling that while constructing the response is a forbidden black magic.